zcash_primitives/src/transaction/builder.rs

```diff
@@ -650,8 +891,8 @@ mod tests {
         {
             let builder = Builder::new(TEST_NETWORK, tx_height);
             assert_eq!(
-                builder.build(&MockTxProver),
-                Err(Error::ChangeIsNegative(
+                builder.mock_build(),
+                Err(Error::InsufficientFunds(
                     (Amount::zero() - DEFAULT_FEE).unwrap()
                 ))
```

Comment:
Bug: `Error::InsufficientFunds` has the inverse semantics to `Error::ChangeIsNegative`:
```suggestion
                Err(Error::InsufficientFunds(DEFAULT_FEE))
```
This and the subsequent equivalent changes should fix the CI test failure.

Reply:
Ah, yeah, I just changed this and didn't try the tests locally.